oio/blob/registrator.py

```diff
@@ -14,181 +14,273 @@
-            for c in chunk_id:
-                if c not in hexdigits:
-                    self.logger.warn('WARN Not a chunk %s' % path)
-                    return
-            # Action
-            try:
-                with open(path) as f:
-                    meta, _ = read_chunk_metadata(f, chunk_id)
-                    self.action(self, path, f, meta)
-                    success = success + 1
-            except NotFound as e:
-                fail = fail + 1
-                self.logger.info("ORPHAN %s/%s in %s/%s %s",
-                                 meta['content_id'], meta['chunk_id'],
-                                 meta['container_id'], meta['content_path'],
-                                 str(e))
-            except Conflict as e:
-                fail = fail + 1
-                self.logger.info("ALREADY %s/%s in %s/%s %s",
-                                 meta['content_id'], meta['chunk_id'],
-                                 meta['container_id'], meta['content_path'],
-                                 str(e))
-            except Exception as e:
-                fail = fail + 1
-                self.logger.warn("ERROR %s/%s in %s/%s %s",
-                                 meta['content_id'], meta['chunk_id'],
-                                 meta['container_id'], meta['content_path'],
-                                 str(e))
-            count = count + 1
-
-            # TODO(jfs): do the throttling
-
-            # periodical reporting
-            t = now()
-            if t - last_report > self.report_interval:
-                self.logger.info("STEP %d ok %d ko %d",
-                                 count, success, fail)
-
-        self.logger.info("FINAL %s %d ok %d ko %d",
-                         self.volume, count, success, fail)
-
-    def _check_chunk(self, path, f, meta):
+        # speed
+        self.chunks_run_time = 0
+        self.max_chunks_per_second = int_value(
+            self.conf.get('chunks_per_second'),
+            self.DEFAULT_CHUNKS_PER_SECOND)
+
+        # counters
+        self.chunks_processed = 0
+        self.chunk_errors = 0
+        self.beans_processed = dict()
+        self.bean_successes = dict()
+        self.bean_already_exists = dict()
+        self.bean_orphans = dict()
+        self.bean_errors = dict()
+        for bean_type in self.BEAN_TYPES:
+            self.beans_processed[bean_type] = 0
+            self.bean_successes[bean_type] = 0
+            self.bean_already_exists[bean_type] = 0
+            self.bean_orphans[bean_type] = 0
+            self.bean_errors[bean_type] = 0
+
+        # report
+        self.start_time = 0
+        self.last_report = 0
+        self.report_interval = int_value(
+            conf.get('report_interval'),
+            self.DEFAULT_REPORT_INTERVAL)
+
+        self.client = ContainerClient(
+            {'namespace': self.namespace}, logger=self.logger)
+        self.ctime = int(time.time())
+
+    def _beans_from_meta(self, meta):
```

Comment:
Maybe it could be factorized with `meta2beans` in first commit ?

Reply:
It's the same method, but I've renamed it.